src/Modules/ContentBlocks/Domain/ContentBlock/ContentBlock.php

```diff
@@ -0,0 +1,177 @@
+
+#[ORM\Entity(repositoryClass: ContentBlockRepository::class)]
+#[ORM\Table(name: 'contentblocks__contentblock')]
+#[DataGrid('ContentBlock')]
+#[ORM\HasLifecycleCallbacks]
+#[DataGridActionColumn(
+    route: 'backend_action',
+    routeAttributes: [
+        'module' => 'content_blocks',
+        'action' => 'content_block_edit',
+    ],
+    routeAttributesCallback: [self::class, 'dataGridEditLinkCallback'],
+    label: 'lbl.Edit',
+    class: 'btn btn-primary btn-sm',
+    iconClass: 'fa fa-edit',
+    requiredRole: ModuleAction::ROLE_PREFIX . 'CONTENT_BLOCKS__CONTENT_BLOCK_EDIT',
+    columnAttributes: ['class' => 'fork-data-grid-action'],
+)]
+class ContentBlock
+{
+    use Blameable;
+    use EntityWithLocaleTrait;
+    use EntityWithSettingsTrait;
+
+    const DEFAULT_TEMPLATE = 'Default.html.twig';
+
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column(name: 'revision_id', type: 'integer')]
+    private int $revisionId;
+
+    #[ORM\Column(type: 'integer')]
+    private int $id;
+
+    #[ORM\Column(name: 'extra_id', type: 'integer')]
+    private int $extraId;
+
+    #[ORM\Column(type: 'string', options: ['default' => self::DEFAULT_TEMPLATE])]
+    private string $template;
+
+    #[ORM\Column(type: 'string')]
+    #[DataGridPropertyColumn(
+        sortable: true,
+        label: 'lbl.Title',
+        route: 'backend_action',
+        routeAttributes: [
+            'module' => 'content_blocks',
+            'action' => 'content_block_edit',
+        ],
+        routeAttributesCallback: [self::class, 'dataGridEditLinkCallback'],
+        routeRole: ModuleAction::ROLE_PREFIX . 'CONTENT_BLOCKS__CONTENT_BLOCK_EDIT',
+        columnAttributes: ['class' => 'title'],
+    )]
+    private string $title;
+
+    #[ORM\Column(type: 'text')]
+    private string $text;
+
+    #[ORM\Column(name: 'hidden', type: 'boolean', options: ['default' => false])]
+    private bool $isHidden;
+
+    #[ORM\Column(type: 'string', enumType: Status::class, options: ['default' => 'active'])]
+    private Status $status;
+
+    private function __construct(ContentBlockDataTransferObject $dataTransferObject)
+    {
+        $this->id = $dataTransferObject->id;
+        $this->extraId = $dataTransferObject->extraId;
+        $this->template = $dataTransferObject->template;
+        $this->locale = $dataTransferObject->locale;
+        $this->title = $dataTransferObject->title;
+        $this->text = $dataTransferObject->text;
+        $this->isHidden = !$dataTransferObject->isVisible;
+        $this->status = $dataTransferObject->status;
+        $this->createdBy = $dataTransferObject->createdBy;
+        $this->updatedBy  = $dataTransferObject->updatedBy;
+        $this->settings = $dataTransferObject->settings;
+    }
+
+    public function getRevisionId(): int
+    {
+        return $this->revisionId;
+    }
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function getUserId(): int
```

Comment:
This method is no longer used and can be removed

Reply:
removed: https://github.com/forkcms/forkcms/pull/3555/commits/0eee4916220942e1607775ea492552d6a300601a